avm/res/compute/virtual-machine/tests/e2e/linux.atmg/main.test.bicep

```diff
@@ -0,0 +1,123 @@
+param serviceShort string = 'cvmlinatmg'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    location: location
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    sshDeploymentScriptName: 'dep-${namePrefix}-ds-${serviceShort}'
+    sshKeyName: 'dep-${namePrefix}-ssh-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+// resource sshKey 'Microsoft.Compute/sshPublicKeys@2022-03-01' existing = {
+//   name: sshKeyName
+//   scope: resourceGroup
+// }
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}-${iteration}'
+  params: {
+    location: location
+    name: '${namePrefix}${serviceShort}'
+    adminUsername: 'localAdminUser'
+    imageReference: {
+      publisher: 'Canonical'
+      offer: '0001-com-ubuntu-server-jammy'
+      sku: '22_04-lts-gen2'
+      version: 'latest'
+    }
+    nicConfigurations: [
+      {
+        ipConfigurations: [
+          {
+            name: 'ipconfig01'
+            pipConfiguration: {
+              publicIpNameSuffix: '-pip-01'
+              tags: {
+                'hidden-title': 'This is visible in the resource name'
+                Environment: 'Non-Prod'
+                Role: 'DeploymentValidation'
+              }
+            }
+            zones: [
+              '1'
+              '2'
+              '3'
+            ]
+            subnetResourceId: nestedDependencies.outputs.subnetResourceId
+          }
+        ]
+        nicSuffix: '-nic-01'
+        tags: {
+          'hidden-title': 'This is visible in the resource name'
+          Environment: 'Non-Prod'
+          Role: 'DeploymentValidation'
+        }
+      }
+    ]
+    osDisk: {
+      diskSizeGB: '128'
+      managedDisk: {
+        storageAccountType: 'Premium_LRS'
+      }
+    }
+    osType: 'Linux'
+    vmSize: 'Standard_DS2_v2'
+    configurationProfile: '/providers/Microsoft.Automanage/bestPractices/AzureBestPracticesProduction'
+    disablePasswordAuthentication: true
+    publicKeys: [
+      {
+        keyData: nestedDependencies.outputs.SSHKeyPublicKey
+        path: '/home/localAdminUser/.ssh/authorized_keys'
+      }
+    ]
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
```

Comment:
```suggestion
```
We should remove anything that isn't needed for this test to test / showcase it's purpose (auto-manage)